pkg/replication/replication.go

```diff
@@ -39,7 +39,7 @@ func (r Replicator) CheckPreparers() error {
 // wish to lock
 func (r Replicator) LockHosts(lock kp.Lock, overrideLock bool) error {
 	for _, host := range r.Nodes {
-		lockPath := kp.LockPath(host, r.Manifest.ID())
+		lockPath := kp.LockPath(kp.IntentPath(host, r.Manifest.ID()))
```

Comment:
With this modification, any binary compiled before this change will not recognize the locks created by binaries compiled after this change, and vice versa.  Do you see this being a problem in production?  (E.g., how many things do we need to simultaneously redeploy and restart for this to not cause problems?)